src/delfino/commands/test.py

```diff
@@ -28,45 +32,67 @@ def _run_tests(app_context: AppContext, name: str, maxfail: int, debug: bool) ->
 
     print_header(f"️Running {name} tests️", icon="🔎🐛")
     ensure_reports_dir(delfino)
-    run(
-        list(
-            filter(
-                None,
-                [
-                    "pytest",
-                    "--cov",
-                    delfino.sources_directory,
-                    "--cov-report",
-                    f"xml:{delfino.reports_directory / f'coverage-{name}.xml'}",
-                    "--cov-branch",
-                    "-vv",
-                    "--maxfail",
-                    str(maxfail),
-                    "-s" if debug else "",
-                    delfino.tests_directory / name,
-                ],
+
+    options: ArgsList = []
+    if debug:
+        options.append("--capture=no")
+        warnings.warn(
+            DeprecatedWarning(
+                "--debug for test-unit and test-integration command",
+                "0.19.0",
+                "1.0.0",
+                "Use --capture=no or -s instead.",
             )
-        ),
+        )
+
+    args: ArgsList = [
+        "pytest",
+        "--cov",
+        delfino.sources_directory,
+        "--cov-report",
+        f"xml:{delfino.reports_directory / f'coverage-{name}.xml'}",
+        "--cov-branch",
+        "-vv",
+        *options,
+        *(extra_options if extra_options else []),
+        delfino.tests_directory / name,
+    ]
+    run(
+        args,
         env_update={"COVERAGE_FILE": delfino.reports_directory / f"coverage-{name}.dat"},
         on_error=OnError.ABORT,
     )
 
 
+def test_options(func):
+    """Common option for test commands."""
+    options = [
+        click.option(
+            "--debug",
+            is_flag=True,
+            help=(
+                "Deprecated. Use --capture=no or -s instead."
+                " | Disables capture, allowing debuggers like `pdb` to be used."
+            ),
+        ),
+        pass_through_option("pytest"),
```

Comment:
Mixing two styles of attaching options. This one is a group of options applied as one decorator. But individual command also have `@pass_app_context`. So either move `@pass_app_context` here as well or use individual decorators at each command.

Reply:
Thanks 👍 . fixed with [fde29c7](https://github.com/radeklat/delfino/pull/35/commits/fde29c7220043abb1a06fe1e013893e8e1c50afe).